### PR TITLE
Auth0 Architecture: Running In Multiple Cloud Providers And Regions

### PR DESCRIPTION
**Pull request by dan-auth0**

### CODE REVIEW

File: _posts/2018-08-12-auth0-architecture-running-in-multiple-cloud-providers-and-regions.markdown [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+Auth0 provides authentication, authorization, and single sign-on services for apps of any type (mobile, web, native) on any stack. Authentication is critical for the vast majority of apps. We designed Auth0 from the beginning so that it could run anywhere: on our cloud, on your cloud, or even on your own private infrastructure.
+
+In this post, we'll talk more about our public SaaS deployments and provide a brief introduction to the infrastructure behind [auth0.com](https://auth0.com/) and the strategies we use to keep it up and running with high availability. This is an updated version of the [2014 High Scalability post about Auth0's architecture](http://highscalability.com/blog/2014/12/1/auth0-architecture-running-in-multiple-cloud-providers-and-r.html).
+
+A lot has changed since then in Auth0. These are some of the highlights:
+
+- We went from processing a couple of million logins per month to 1.5+ billion logins per month, serving thousands of customers, including FuboTV, Mozilla, JetPrivilege, and more.
+
+- We implemented new features like [_custom domains_](https://auth0.com/docs/custom-domains) and improved existing ones.
```
Comment (dirceu) [inline _posts/2018-08-12-auth0-architecture-running-in-multiple-cloud-providers-and-regions.markdown]: What about something like this?

> We implemented new features like [_custom domains_](https://auth0.com/docs/custom-domains), scaled [bcrypt operations](https://auth0.engineering/bcrypt-as-a-service-9e71707bda47), vastly improved [user search](https://auth0.com/docs/users/search/v3), and much more.

Comment (dan-auth0) [reply _posts/2018-08-12-auth0-architecture-running-in-multiple-cloud-providers-and-regions.markdown]: LGTM!

File: _posts/2018-08-12-auth0-architecture-running-in-multiple-cloud-providers-and-regions.markdown [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  - aws
+  - mongoDB
+  - testing
+  - automation
+  - cdn
+related:
+  -
+---
+
+Auth0 provides authentication, authorization, and single sign-on services for apps of any type (mobile, web, native) on any stack. Authentication is critical for the vast majority of apps. We designed Auth0 from the beginning so that it could run anywhere: on our cloud, on your cloud, or even on your own private infrastructure.
```
Comment (scottmathson-auth0) [inline _posts/2018-08-12-auth0-architecture-running-in-multiple-cloud-providers-and-regions.markdown]: `could run anywhere: on our cloud, *on your cloud*, or even on your own private infrastructure.` 🤔

File: _posts/2018-08-12-auth0-architecture-running-in-multiple-cloud-providers-and-regions.markdown [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+In this post, we'll talk more about our public SaaS deployments and provide a brief introduction to the infrastructure behind [auth0.com](https://auth0.com/) and the strategies we use to keep it up and running with high availability. This is an updated version of the [2014 High Scalability post about Auth0's architecture](http://highscalability.com/blog/2014/12/1/auth0-architecture-running-in-multiple-cloud-providers-and-r.html).
+
+A lot has changed since then in Auth0. These are some of the highlights:
+
+- We went from processing a couple of million logins per month to 1.5+ billion logins per month, serving thousands of customers, including FuboTV, Mozilla, JetPrivilege, and more.
```
Comment (scottmathson-auth0) [inline _posts/2018-08-12-auth0-architecture-running-in-multiple-cloud-providers-and-regions.markdown]: Internal linking opportunities here.

FuboTV: https://auth0.com/learn/sports-centric-streaming-service-fubotv-sees-50-roi-just-auth0s-security/
Mozilla: https://auth0.com/blog/auth0-mozilla-partnership/
JetPrivilege: https://auth0.com/learn/jetprivilege-case-study/

File: _posts/2018-08-12-auth0-architecture-running-in-multiple-cloud-providers-and-regions.markdown [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+- Lower maintenance costs.
+
+- Easier configuration.
+
+There are a few downsides, like the fact that we need to run Lambdas to perform some configurations (like adding custom headers to PDF files and things like that). Still, the upsides definitely make up for that.
+
+## Extend
+
+One of the features we provide is the ability to run custom code as part of the login transaction, either via [_authentication rules_](https://auth0.com/docs/rules/current) or [_custom database connections_](https://auth0.com/docs/connections/database/custom-db). These features are powered by [_Extend_](https://auth0.com/extend), an extensibility platform that grew out of Auth0 and is now being used by other companies as well. With Extend, our customers can write anything they want in those scripts and rules, allowing them to extend profiles, normalize attributes, send notifications, and much more.
```
Comment (scottmathson-auth0) [inline _posts/2018-08-12-auth0-architecture-running-in-multiple-cloud-providers-and-regions.markdown]: Updated Extend url to go to where it redirects to: https://goextend.io/

File: _posts/2018-08-12-auth0-architecture-running-in-multiple-cloud-providers-and-regions.markdown (hunk 2 above)
Comment (dan-auth0) [reply _posts/2018-08-12-auth0-architecture-running-in-multiple-cloud-providers-and-regions.markdown]: I think here we prefer not to stress any option, as to communicate that all options are available.

**Pull request merged**